fastlane/lib/fastlane/commands_generator.rb

```diff
@@ -42,6 +47,26 @@ def self.start
       end
     end
 
+    def self.confirm_troubleshoot
+      if Helper.is_ci?
+        UI.error "---"
+        UI.error "You are trying to use '--troubleshoot' on CI"
+        UI.error "this option is not usable in CI, as it is insecure"
+        UI.error "---"
+        UI.user_error!("Do not use --troubleshoot in CI")
```

Comment:
Do we really want to force people not to do that? Showing a warning is fine, but I think we should still let them do their thing.